mxcube3/routes/qutils.py

```diff
@@ -1422,6 +1422,17 @@ def queue_model_child_added(parent, child):
             parent_entry.enqueue(dc_entry)
             sample = parent.get_parent()
 
+            sampleID = sample._node_id
```

Comment:
Cant we attach the shape in the corresponding queue entry ?, I think that would be a better solution.

Reply:
but the queue_entry does not have any info about the shapes. I though about it, but it would meant o modify the queue_entry only for this specific case.

Reply:
Hm the DataCollectionModel should have information about its CenterdPosition, I think you can get the shape ID from there. Or just copy the entire CenteredPosition object over to the new collection.

It should be something like model.acquisitions[0].acquisition_parameters.centred_position

Maybe it helps ?

Cheers,
Marcus

Reply:
I will have a look... no promises...

Reply:
the issue is that we need the shape_id, I can copy the raw cpos object from the char entry, but it does not contain the shape_id, only motor positions (in fact it is already created). I could also add the shapes_hwobj and retrieve the id from there (comparing dicts?), but in this case the cpos from the diff plan entry would contain motor positions + shape_id, while the originated char entry does not.

Maybe the best thing would be to add the shape_id whenever a cpos is created, thus before the diff plan is created. but for the purpose of this PR I would suggest to accept as it is. What do you think?